Nova menus: check that the term exists before trying to access a property in it.

eliorivero opened the pull request:
This PR:

- avoids displaying a bunch of PHP notices in the menu list view in admin when no terms exist.

- the UP/DOWN buttons now work as expected, moving the group of items upwards or downwards. Previously they led to a blank page asking if user wanted to do it and they never work anyway.

- while editing an item, the first category in the list can now be deselected _if and when_ there is another category that user wants to select. Previously, you couldn't deselect the first category, even if there was another category already checked.

fixes #3407

jeherve commented:
I can't seem to be able to reproduce the notices on `master`, when following the instructions you posted in #3407.

I get the error when clicking on "Up", though, even with your branch.

Is there anything else I could try to reproduce the notices?

eliorivero commented:
@jeherve menus were tested with TwentyTwelve last time and now with TwentySixteen. The issue with UP/DOWN buttons is now fixed. I also fixed a JS issue where the first category in the category list when editing a menu item couldn't be unchecked even if there was another category checked already. The menu original idea was to have one and only one category applied all the time, but this made it impossible to switch to a different category.

jeherve commented:
This seems to work in my tests. LGTM!
